androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/stopDetails/StopDetailsViewModelTest.kt

```diff
@@ -902,24 +883,21 @@ class StopDetailsViewModelTest {
                     pinnedRoutes = setOf(),
                     updateStopFilter = { _, _ -> },
                     updateTripFilter = { _, _ -> },
-                    setMapSelectedVehicle = {},
-                    coroutineDispatcher = dispatcher
+                    setMapSelectedVehicle = {}
                 )
             }
         }
 
-        advanceUntilIdle()
+        composeTestRule.waitForIdle()
 
         composeTestRule.waitUntil {
-            // In resumed state, so joined 1 time for stop, 1 time for resume.
-            // Need to start with lifecycle resumed in order to test pause
-            stopPredictionsConnectCount == 2
-
-            // Trip channels rejoin is no-op because tripData hasn't been set yet
-            && tripPredictionsConnectCount == 1 && vehicleConnectCount == 1
+            // resume rejoin will have been skipped
+            stopPredictionsConnectCount == 1 &&
+                tripPredictionsConnectCount == 1 &&
+                vehicleConnectCount == 1
```

Comment:
“tripData hasn’t been set yet” wasn’t actually guaranteed to be true: if the `clearAndLoadTripDetails` call finished before the `LifecycleResumeEffect` kicked in, tripData would in fact have been set, causing the trip channels to be joined a second time as well.

It seemed to me like the easiest approach here was to just not rejoin the channels in the `LifecycleResumeEffect` until we’ve actually been to the background first, so that it doesn’t matter when the `LifecycleResumeEffect` wakes up; this race condition can only hit if `loadTripDetails` is instant, which it won’t be in the real app, so this was only ever a test issue, but the new behavior still feels more correct on principle.